### DISCUSSION TITLE
How to add an optional field to the user schema ?

### DISCUSSION THREAD

**Discussion by rubalo** [fastapi-users/fastapi-users#1307] [Q&A]
Hello, 

I'm trying to add an optional field to the user schema but it doesn't end up in the database. I would like to know what's the best way to approach this.

```
class UserCreate(schemas.BaseUserCreate):
    is_member: bool = False
```

This field is never put in the database because of the create_user() [function from the user_manager](https://github.com/fastapi-users/fastapi-users/blob/ff9fae631cdae00ebc15f051e54728b3c8d11420/fastapi_users/manager.py#L135)

```
user_dict = (
     user_create.create_update_dict()
     if safe
     else user_create.create_update_dict_superuser()
)
```

Pydantic detect this field as not set and thus the field is not passed to the user_db create.

Maybe I get it wrong and my approach is bad, otherwise I have a few tweaks in mind but I wanted to know what do you think first.

My 2 cents would go to "You can override the function", but I would like to know the idea behind not having the exclude_unset attribute to False in the create_update_dict function.

Cheers, Jerome

**Comment by frankie567** [comment]
The problem probably doesn't come from the Pydantic schema, but rather by your underlying database model.

You'll need to add this field to your [SQLAlchemy model](https://fastapi-users.github.io/fastapi-users/12.1/configuration/databases/sqlalchemy/#create-the-user-model) or [Beanie model](https://fastapi-users.github.io/fastapi-users/12.1/configuration/databases/beanie/#create-the-user-model), depending on the DB backend you use.

**Reply by rubalo** [reply]
Hi, thanks for your answer, as I mentioned, this comes from the create_update_dict() and create_update_dict_superuser() functions and the underlying call to the pydantic model_dump function with the exclude_unset parameter set to True.

**Reply by frankie567** [reply]
Hmm, so what you want is the `is_member` to be set to `False` on user registration? But it shouldn't be set by the end user?

If that's the case, I would suggest to put this default directly on the DB model.

**Reply by rubalo** [reply]
Well I saw another discussion where the default values on a SqlAlchemy models are not advised.
What I did is override the user_manager.create function and replace the create_update_dict() function : 
```
user_dict = (
     user_create.create_update_dict()
     if safe
     else user_create.create_update_dict_superuser()
)
```

been replaced by 

```
user_dict = (
            user_create.model_dump(
                exclude_unset=False,
                exclude={
                    "id",
                    "is_superuser",
                    "is_active",
                    "is_verified",
                    "oauth_accounts",
                },
            )
            if safe
            else user_create.model_dump(exclude_unset=False, exclude={"id"})
        )
```

Maybe there is a better way...
Thanks for your help
